### PR TITLE
completed task #7. GET /api/articles/:article_id (comment count)

### PR DESCRIPTION
**Pull request by baurzhaan**
completed tasks 7

### CODE REVIEW

File: __tests__/news.test.js [hunk 1]
```diff
@@ -31,7 +31,7 @@ describe('GET /api/topics', () => {
 });
 
 describe('GET /api/articles/:article_id', () => {
-  test('responds the article with an id of 1', () => {
+  test.skip('responds the article with an id of 1', () => {
```
Comment (emilyb93) [inline __tests__/news.test.js:34]: do not skip your tests. Remove this and then push up so i know this is working correctly

File: models/articles.models.js [hunk 2]
```diff
@@ -2,9 +2,16 @@ const db = require('../db/connection.js');
...
 exports.selectArticleById = (articleId) => {
   return db.query('SELECT * FROM articles WHERE article_id = $1;', [articleId])
-  .then(({ rows }) => {
-    if (rows.length) return rows[0];
+  .then(({ rows: articleRows }) => {
+    if (articleRows.length) return articleRows[0];
     return Promise.reject({ code: 404, msg: 'The article not found' });
+  })
+  .then((article) => {
+    return db.query('SELECT COUNT(*) FROM comments WHERE article_id = $1;', [articleId])
```
[the remark this answers is not in the document]
Comment (baurzhaan) [reply models/articles.models.js]: done

File: __tests__/news.test.js [hunk 3]
```diff
@@ -152,4 +152,111 @@ describe('GET /api/users', () => {
...
+    });
+  })
+  test('404: Returns \'Route not found\' when the route doesn\'t exist', () => {
+    return request(app)
+    .get('/api/nothing')
+    .expect(404)
+    .then((response) => {
+      expect(response.body.msg).toBe('Route not found');
+    });
+  });
```
Comment (sirKevlar) [inline __tests__/news.test.js:224]: Your original 404 test for bad path has already covered this and will handle all bad paths (unless there is a parametric endpoint involved) No need to repeat tests for a straight bad path

File: __tests__/news.test.js [hunk 4]
```diff
@@ -152,4 +152,111 @@ describe('GET /api/users', () => {
...
+        expect(body.msg).toBe('The article not found');
+      });
+  });
+  test('400: responds with message \'Bad request\' when the article id is not valid', () => {
+    return request(app)
+      .get('/api/articles/not_valid_request')
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe('Not valid request');
+      });
```
Comment (sirKevlar) [inline __tests__/news.test.js:260]: This should also be a separate PR for ticket #9

File: app.js [hunk 5]
```diff
@@ -1,35 +1,41 @@
...
 app.use((error, _, response, next) => {
-  if (error.code === '22P02') {
-    response.status(400).send({ msg: 'Not valid request' });
+  if (error.code === 404 && error.msg === 'The article not found') {
+    response.status(error.code).send({ msg: error.msg });
   } else {
     next(error);
-  }
-})
+  };
```
Comment (sirKevlar) [inline app.js:29]: At the moment, this custom error handler will only work for 404 errors... Could we make it more generic?

File: __tests__/news.test.js [hunk 6]
```diff
@@ -152,4 +152,111 @@ describe('GET /api/users', () => {
...
+    return request(app)
+      .get('/api/articles/3')
+      .expect(200)
+      .then(({ body }) => { // destructure body from the response
+        expect(body).toBeInstanceOf(Object);
+        expect(body).toEqual(articleWithCommentCount);
+    });
+  });
+});
+
```
[the remark this answers is not in the document]
Comment (baurzhaan) [reply __tests__/news.test.js:178]: done. removed unnecessary properties check.

File: __tests__/news.test.js [hunk 7]
```diff
@@ -152,4 +152,111 @@ describe('GET /api/users', () => {
...
+  test('404: Returns \'Route not found\' when the route doesn\'t exist', () => {
+    return request(app)
+    .get('/api/nothing')
+    .expect(404)
+    .then((response) => {
+      expect(response.body.msg).toBe('Route not found');
+    });
+  });
+});
+
```
[the remark this answers is not in the document]
Comment (baurzhaan) [reply __tests__/news.test.js:226]: sorry for that, will keep it separate. I was under impression that I pulled only the ticket 7. 

Yesterday at the end of the day I got a few comments from Emily on ticket 7, then I fixed them, but it was too late to pull request before 5 pm. After, in the evening I did tickets 8 & 9, pushed them (but didn’t pull the request). So, it seems that when I push under the same branch, then it automatically goes into the current branch. Will keep that in mind!.

File: __tests__/news.test.js (hunk 3 above)
Comment (baurzhaan) [reply __tests__/news.test.js:224]: got that. We have this test case in /api/topics, so no need in "/api/users" and  "/api/articles". Removed excessive tests.

File: app.js [hunk 8]
```diff
@@ -1,35 +1,41 @@
...
+});
```
[the remark this answers is not in the document]
Comment (baurzhaan) [reply app.js:30]: )) we had slightly debated about it. As a result, I've replaced '_' by '_request', so I (and hopefully reader of my code) understand that this parameter is not  used.

File: app.js (hunk 5 above)
Comment (baurzhaan) [reply app.js:29]: done. It took me a while to fix it )
